Sources/Hummingbird/Server/Request.swift

```diff
@@ -100,9 +119,29 @@ public struct HBRequest: Sendable, HBSendableExtensible {
     public func decode<Type: Decodable>(as type: Type.Type) throws -> Type {
         do {
             return try self.application.decoder.decode(type, from: self)
-        } catch {
+        }
+        catch DecodingError.dataCorrupted(_) {
+            let message = "The given data was not valid input."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.keyNotFound(key, _) {
+            let path = key.pathKeyValue
+            let message = "Coding key `\(path)` not found."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.valueNotFound(_, context) {
+            let path = context.codingPath.pathKeyValue
+            let message = "Value not found for `\(path)` key."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch let DecodingError.typeMismatch(type, context)  {
+            let path = context.codingPath.pathKeyValue
+            let message = "Type mismatch for `\(path)` key, expected `\(type)` type."
+            throw HBHTTPError(.badRequest, message: message)
+        }
+        catch {
```

Comment:
Should you catch `HBHTTPError` here and then just throw it again. Otherwise if you do catch it you are throwing a `HBHTTPError` that has a message that describes the `HBHTTPError` you just caught. In actual fact to be correct you should catch all errors that conform to `HBHTTPResponseError`
```
catch let error as HBHTTPResponseError {
    throw error
}

Reply:
Is this how you imagined? 🤔